### PR TITLE
Feature/lstm

### PR DESCRIPTION
**Pull request by dzhwinter**
add static rnn benchmark scripts (book chapter6)
fix https://github.com/PaddlePaddle/Paddle/issues/6156

### CODE REVIEW

File: paddle/understand_sentiment_lstm.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        default='GPU',
+        choices=['CPU', 'GPU'],
+        help='The device type.')
+    parser.add_argument(
+        '--infer_only', action='store_true', help='If set, run forward only.')
+    parser.add_argument(
+        '--use_cprof', action='store_true', help='If set, use cProfile.')
+    parser.add_argument(
+        '--use_nvprof',
+        action='store_false',
```
Comment (qingqing01) [inline paddle/understand_sentiment_lstm.py]: store_false -> store_ture  . disable use_nvprof by defalut.

File: paddle/understand_sentiment_lstm.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    emb_dim = args.emb_dim
+    seq_len = args.seq_len
+
+    emb = fluid.layers.embedding(input=data, size=[dict_dim, emb_dim])
+    emb = fluid.layers.reshape(x=emb, shape=[batch_size, seq_len, args.emb_dim])
+    emb = fluid.layers.transpose(x=emb, axis=[1, 0, 2])
+
+    c_pre_init = fluid.layers.fill_constant(
+        dtype=emb.dtype, shape=[batch_size, emb_dim], value=0.0)
+    layer_1_out = fluid.layers.lstm(
```
Comment (qingqing01) [inline paddle/understand_sentiment_lstm.py]: Maybe we can add an argument to allow to set multi-lstm？Here

```python
lstm_var = emb
for i in range(arg.lstm_num):
    lstm_var = fluid.layers.lstm(lstm_var)
```

Comment (qingqing01) [inline paddle/understand_sentiment_lstm.py]: Maybe we can add an argument to allow to set multi-lstm？Here

```python
lstm  = emb
for i in range(arg.lstm_num):
    lstm = fluid.layers.lstm(lstm)
```

Comment (dzhwinter) [reply paddle/understand_sentiment_lstm.py]: fixed.

File: paddle/understand_sentiment_lstm.py (hunk 1 above)
Comment (dzhwinter) [reply paddle/understand_sentiment_lstm.py]: done.

File: paddle/understand_sentiment_lstm.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    emb = fluid.layers.reshape(x=emb, shape=[batch_size, seq_len, args.emb_dim])
+    emb = fluid.layers.transpose(x=emb, axis=[1, 0, 2])
+
+    c_pre_init = fluid.layers.fill_constant(
+        dtype=emb.dtype, shape=[batch_size, emb_dim], value=0.0)
+    layer_1_out = emb
+    for i in range(stacked_num):
+        layer_1_out = fluid.layers.lstm(
+            layer_1_out, c_pre_init=c_pre_init, hidden_dim=emb_dim)
+        layer_1_out = fluid.layers.transpose(x=layer_1_out, axis=[1, 0, 2])
```
Comment (qingqing01) [inline paddle/understand_sentiment_lstm.py]: The `fluid.layers.transpose` should be in the outside of for loop.

Comment (dzhwinter) [reply paddle/understand_sentiment_lstm.py]: fixed.

**Pull request merged**